lib/Elastica/Query/FunctionScore.php

```diff
@@ -147,7 +148,8 @@ public function addDecayFunction(
         $offset = null,
         $decay = null,
         $weight = null,
-        $filter = null
```

Comment:
I think the addDecayFunction is becoming too many params. We should probably switch to a `$params` array to not have BC breaks in case one of the params disappears. Also this makes adding / non supported params easier.